soda/core/soda/execution/data_source.py

```diff
@@ -242,6 +247,154 @@ def sql_information_schema_identifier(self) -> str:
+        if is_final:
+            sql += f"\n select * from {cte_name}"
+        return sql
+
+    def profiling_sql_value_frequencies_select(self) -> str:
+        return dedent(
+            f"""
+            , mins as (
+            select value_name, row_number() over(order by value_name asc) as idx, frequency, 'mins' as metric_name
+            from freq_values
+            where value_name is not null
+            order by value_name asc
+            limit 5
+        )
+        , maxes as (
+
+            select value_name, row_number() over(order by value_name desc) as idx, frequency, 'maxes' as metric_name
+            from freq_values
+            where value_name is not null
+            order by value_name desc
+            limit 5
+        )
+        {self.profiling_sql_frequent_values_cte(source_table_name='freq_values', cte_name='frequent_values', is_final=False)}
+        , final as (
+            select
+                mins.value_name as mins
+                , maxes.value_name as maxes
+                , frequent_values.value_name as frequent_values
+                , frequent_values.frequency as frequency
+            from mins
+            join maxes
+                 on mins.idx = maxes.idx
+            join frequent_values
+                on mins.idx = frequent_values.idx
+            order by frequency desc
+        )
+        select * from final
+            """
+        )
+
+    def profiling_sql_numeric_aggregates(self, table_name: str, column_name: str) -> str:
+        column_name = self.quote_column(column_name)
+        return dedent(
+            f"""
+            select
+                avg({column_name}) as average
+                , sum({column_name}) as sum
+                , variance({column_name}) as variance
+                , stddev({column_name}) as standard_deviation
+                , count(distinct({column_name})) as distinct_values
+                , sum(case when {column_name} is null then 1 else 0 end) as missing_values
+            from {table_name}
+            """
+        )
+
+    def profiling_sql_text_aggregates(self, table_name: str, column_name: str) -> str:
+        column_name = self.quote_column(column_name)
+        return dedent(
+            f"""
+            select
+                count(distinct({column_name})) as distinct_values
+                , sum(case when {column_name} is null then 1 else 0 end) as missing_values
+                , avg(length({column_name})) as avg_length
+                , min(length({column_name})) as min_length
+                , max(length({column_name})) as max_length
+            from {table_name}
+            """
+        )
+
+    def histogram_sql_and_boundaries(
+        self, table_name: str, column_name: str, min: int | float, max: int | float
+    ) -> tuple[str, list[int | float]]:
+        # TODO: make configurable or derive dynamically based on data quantiles etc.
+        number_of_bins: int = 20
+
+        assert (
+            min < max
```

Comment:
What happens if column contains only one value or min and max are the same?

Reply:
To be honest, I didn't really question the code as I took it from soda-sql but I see that it could indeed be an issue and I'll add some handling for this.

Reply:
Since in other places we use capitals already - I think it would be good to keep the same - because these queries end up in the log and can be a bit annoying for the users if the formatting is inconsistent.

Reply:
Fair enough :) I haven't checked soda-sql code - since this is much more focused PR it was easy to spot.

Reply:
@vijaykiran on this one, we're actually passing the min and the max that we will have obtained from aggregates query before so there will always be a value or sorts.

If this assertion fails (and it must be true) then I guess histograms will not run.

I think it may be better to raise a log error rather than throw an assertion correct?